app/services/PrepareUploadService.scala

```diff
@@ -47,12 +47,17 @@ class PrepareUploadService @Inject()() {
         Seq(
           Json.arr(
             "content-length-range",
-            settings.minimumFileSize,
-            settings.maximumFileSize
+            settings.minimumFileSize.foldLeft(PrepareUploadService.minFileSize)(math.max),
```

Comment:
Just I wonder if it's not a bit too cryptic. But replacing it with something more verbose would require much more of code. Like this:

settings.minimumFileSize match {
case None => PrepareUploadService.minFileSize
case Some(customMinFileSize) => customMinFileSize max PrepareUploadService.minFileSize
}